timesketch/tsctl.py

```diff
@@ -639,3 +645,99 @@ def validate_context_links_conf(path):
             print(f'=> OK: "{entry}"')
         except (ValidationError, SchemaError) as err:
             print(f'=> ERROR: "{entry}" >> {err}\n')
+
+
+# Analyzer stats cli command
+@cli.command(name="analyzer-stats")
+@click.argument(
+    "analyzer_name",
+    required=False,
+    default="all",
+)
+@click.option(
+    "--timeline_id",
+    required=False,
+    help="Timeline ID if the analyzer results should be filtered by timeline.",
+)
+@click.option(
+    "--scope",
+    required=False,
+    help="Scope on: [many_hits, long_runtime, recent]",
+)
+@click.option(
+    "--result_text_search",
+    required=False,
+    help="Search in result text. E.g. for a specific rule_id.",
+)
+@click.option(
+    "--limit",
+    required=False,
+    help="Limit the number of results.",
+)
+def analyzer_stats(analyzer_name, timeline_id, scope, result_text_search, limit):
+    """Prints analyzer stats."""
+
+    if timeline_id:
+        timeline = Timeline.query.get(timeline_id)
+        if not timeline:
+            print("No timeline found with this ID.")
+            return
+        if analyzer_name == "all":
+            analysis_history = Analysis.query.filter_by(timeline=timeline).all()
+        else:
+            analysis_history = Analysis.query.filter_by(
+                timeline=timeline, analyzer_name=analyzer_name
+            ).all()
+    elif analyzer_name == "all":
+        analysis_history = Analysis.query.filter_by().all()
+    else:
+        # analysis filter by analyzer_name
+        analysis_history = Analysis.query.filter_by(analyzer_name=analyzer_name).all()
+
+    df = pd.DataFrame()
+    for analysis in analysis_history:
+        # extract number of hits from result to a int so it could be sorted
+        try:
+            matches = int(re.search(r"\d+(?=\s+events)", analysis.result))
+        except TypeError:
+            matches = 0
+        new_row = pd.DataFrame(
+            [
+                {
+                    "runtime": analysis.updated_at - analysis.created_at,
+                    "hits": matches,
+                    "timeline_id": analysis.timeline_id,
+                    "result": analysis.result,
```

Comment:
Since the result string can be arbitrarily long, I would recommend to place it to the end of the table/columns.

Reply:
done